WordPress/Classes/ViewRelated/Reader/ReaderDetailCoordinator.swift

```diff
@@ -156,6 +165,23 @@ class ReaderDetailCoordinator {
         viewController?.navigationController?.popViewController(animated: true)
     }
 
+    /// Some posts have content from private sites that need special cookies
+    ///
+    /// Use this method to make sure these cookies are downloaded.
+    /// - Parameter webView: the webView where the post will be rendered
+    /// - Parameter completion: a completion block
+    func authenticateIfNeeded(in webView: WKWebView, completion: @escaping () -> Void) {
+        guard let postURLString = post?.permaLink,
+            let postURL = URL(string: postURLString) else {
+            completion()
+            return
+        }
+
+        authenticatedRequest(for: postURL, on: webView) { _ in
```

Comment:
This was really hard to understand at first: we ask for an authenticated request and ignore the result! 🤔

It took me a while to understand.

I stopped to think about this a bit more and realized the method we're calling here does more than just authenticating a request - and we're actually calling it for that side effect, which isn't really documented and isn't self-evident from the method name.

Consider how the code would look if we, instead, had a line like this:

```swift
authenticator.storeAuthenticationCookies(for: postURL, in: webView.configuration.websiteDataStore.httpCookieStore)
```

And then have in Authenticator a new method, looking kind of like this:

```swift
func storeAuthenticationCookies(for url: URL, cookieJar: CookieJar, completion: () -> ()) {
    authenticator.request(url: url, cookieJar: cookieStore) { _ in
        completion()
    }
}
```

I think this would greatly improve the readability of your code.  Just my 2 cents.

Reply:
Loved this feedback, it was indeed confusing.

I changed in 6cdea72021b997c9687cc6939b776912c36a0952

Let me know what do you think: is it more clear?